opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -372,37 +374,57 @@ def set_attribute(self, key: str, value: types.AttributeValue) -> None:
             logger.warning("invalid key (empty or null)")
             return
 
-        if isinstance(value, Sequence):
-            error_message = self._check_attribute_value_sequence(value)
-            if error_message is not None:
-                logger.warning("%s in attribute value sequence", error_message)
-                return
-            # Freeze mutable sequences defensively
-            if isinstance(value, MutableSequence):
-                value = tuple(value)
-        elif not isinstance(value, (bool, str, int, float)):
-            logger.warning("invalid type for attribute value")
+        error_message = self._check_attribute_value(value)
+        if error_message is not None:
+            logger.warning(error_message)
             return
+        # Freeze mutable sequences defensively
+        if isinstance(value, MutableSequence):
+            value = tuple(value)
 
         self.attributes[key] = value
 
     @staticmethod
-    def _check_attribute_value_sequence(sequence: Sequence) -> Optional[str]:
+    def _check_attribute_value(value: types.AttributeValue) -> Optional[str]:
```

Comment:
Does this need to be a static method? It seems like it doesn't depend on the actual `Span` class.